newIDE/app/src/ResourcesList/ResourcePreview/Model3DPreview.js

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import Resource3DPreviewContext from './Resource3DPreviewContext';
+import PlaceholderLoader from '../../UI/PlaceholderLoader';
+import CheckeredBackground from '../CheckeredBackground';
+import { CorsAwareImage } from '../../UI/CorsAwareImage';
+
+const styles = {
+  container: {
+    position: 'relative',
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'stretch',
+    minHeight: 0,
+  },
+  background: {
+    position: 'absolute',
+    top: 0,
+    left: 0,
+    width: '100%',
+    height: '100%',
+    background: 'url("res/transparentback.png") repeat',
+  },
+  screenshot: {
+    position: 'relative',
+    objectFit: 'contain',
+    width: '100%',
+    height: '100%',
+  },
+  loader: {
+    padding: 0,
+  },
+};
+
+type Props = {|
+  modelUrl: string,
+  expand?: boolean,
+  fullWidth?: boolean,
+  size?: number,
+|};
+
+const Model3DPreview = ({ modelUrl, size, expand, fullWidth }: Props) => {
+  const { getResourcePreview } = React.useContext(Resource3DPreviewContext);
+  const [imageDataUrl, setImageDataUrl] = React.useState(modelUrl ? null : '');
+
+  // Load the model preview when the component mounts or when the modelUrl changes.
+  React.useEffect(
+    () => {
+      (async () => {
+        if (!modelUrl) {
+          return;
+        }
+        const dataUrl = await getResourcePreview(modelUrl);
+        setImageDataUrl(dataUrl);
```

Comment:
You could use isMounted hook to check if the component is still mounted after the await. If not, just return without attemption a set state which would trigger a warning.